### PR TITLE
🎉 New Source: Captain Data [Configuration based CDK]

### PR DESCRIPTION
**Pull request by elliottrabac**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

Implement a new Source for Captain Data.

## How
*Describe the solution*

Developed using (Configuration Based Source) low-code CDK.

## Recommended reading order
1. `manifest.yaml`
2. `schemas/*`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

None. New Source only.

## Pre-merge Checklist
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Connector version is set to `0.0.1`
    - [x] `Dockerfile` has version `0.0.1`
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-captain-data/source_captain_data/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  - "#/definitions/job_results_stream"
+
+check:
+  type: CheckStream
+  stream_names:
+    - "workspace"
+
+spec:
+  type: Spec
+  documentationUrl: https://docs.captaindata.co/#1da39f5f-a26f-4a53-9e84-3e4b28077eae
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-captain-data/source_captain_data/manifest.yaml:153]: This is the Airbyte documentationUrl https://docs.airbyte.com/integrations/sources/captain-data

File: airbyte-integrations/connectors/source-captain-data/source_captain_data/schemas/job_results.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-captain-data/source_captain_data/schemas/job_results.json:3]: Please add `additionalProperties: true` to all streams. This make more easy to receive new fields.

File: airbyte-integrations/connectors/source-captain-data/source_captain_data/schemas/workflows.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      "type": ["null", "string"]
+    },
+    "name": {
+      "type": ["null", "string"]
+    },
+    "project_id": {
+      "type": ["null", "integer"]
+    },
+    "steps": {
+      "type": ["null", "array"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-captain-data/source_captain_data/schemas/workflows.json:15]: you need to specify the items inside the array

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments @elliottrabac thanks for the contribution

**Comment by elliottrabac** [comment]
@marcosmarxm thanks for the review. I updated the PR 👍

**Comment by elliottrabac** [comment]
@marcosmarxm 🙏 Any chance to have a quick review here?

**Review by sajarin — CHANGES_REQUESTED** [review]
Thanks for the PR @elliottrabac, in addition to some of the changes requested can you also: 
* create a [manifest.yaml](https://docs.airbyte.com/connector-development/connector-metadata-file) file for this connector? 
* share credentials in order to test this connector with our team? Or share screenshots of passing integration tests? 

This last one is optional but it would really help us get this merged. At the moment, it seems like creating an account with Captain Data would require us to schedule a demo, which may take some time. 

It would be really nice if we could verify the connector is working on our end but if you have screenshots of passing gradle output (via the `./gradlew :airbyte-integrations:connectors:source-captain-data:integrationTest`) command that would suffice.

**Comment by elliottrabac** [comment]
@sajarin what I am supposed to add for `definitionId` in metadata.yml?

**Comment by sajarin** [comment]
> @sajarin what I am supposed to add for `definitionId` in metadata.yml?

definitionId should be a UUID, you can generate one here: https://www.uuidgenerator.net/version4

**Comment by elliottrabac** [comment]
@sajarin I did the suggested changes

I struggle with the gradle setup :/ You can create an account easily here -> https://app.captaindata.co/signup

Also, we are running this connector in production internally as a custom connector (and some of our clients too).
Screenshot 2023-06-13 at 10 28 52
Screenshot 2023-06-13 at 10 29 13

`python main.py read --config secrets/config.json --catalog integration_tests/configured_catalog.json` is working fine:
Screenshot 2023-06-13 at 10 29 29

**Comment by elliottrabac** [comment]
@sajarin let me know if that's okay :)

**Comment by elliottrabac** [comment]
@sajarin any news? Captain Data would love to implement Airbyte Cloud and this connector for their customer as soon as possible :)

**Comment by elliottrabac** [comment]
@sajarin @marcosmarxm sorry for pinging you again

Captain Data would like to suggest their customers to use Airbyte Cloud with this beta connector. Do you have an ETA for this review?

**Comment by marcosmarxm** [comment]
> @sajarin @marcosmarxm sorry for pinging you again
> 
> Captain Data would like to suggest their customers to use Airbyte Cloud with this beta connector. Do you have an ETA for this review?

Sorry the delay and missing reply @elliottrabac. I'll return to your contribution in this week.

**Review by marcosmarxm — APPROVED** [review]
Thanks @elliottrabac the implementation looks correct. I'm going to release the connector during the week, right now our CI is broken.

**Comment by elliottrabac** [comment]
> Thanks @elliottrabac the implementation looks correct. I'm going to release the connector during the week, right now our CI is broken.

Thanks a lot!

**Review by octavia-approvington — APPROVED** [review]

**Comment by elliottrabac** [comment]
@marcosmarxm is the connector available on Airbyte Cloud?

**Pull request merged**